workflow/common/util.go

```diff
@@ -290,9 +314,58 @@ func ProcessArgs(tmpl *wfv1.Template, args wfv1.Arguments, globalParams, localPa
 	}
 	tmpl.Inputs.Artifacts = newInputArtifacts
 
+	//copy passthrough parameters from Inputs to Arguments
+	if tmpl.Steps != nil {
```

Comment:
I think this can be moved to https://github.com/argoproj/argo/blob/d5fe5f981fb112ba01ed77521ae688f8a15f67b9/workflow/controller/steps.go#L150.